### ISSUE TITLE
Testing: Test of restart code needed

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#232]
Clear tests (and therefore examples) of working restarts with VMC and DMC are needed.

Based on these tests, the restart code can be updated/cleaned/rewritten as needed. No need for backward compatibility.

**Comment by prckent** [comment]
@markdewing Delegate/ask for helpers as needed

**Comment by prckent** [comment]
This is also needed to interface with the ECP VeloC project.

**Comment by ye-luo** [comment]
This part of my project for the summer student. Deal with the restart and paralell IO.

**Comment by prckent** [comment]
Tests are due by end of June. Fixes later.

**Comment by prckent** [comment]
Provisionally added as a requirement to the next release (3.1) particularly in light of bug referenced in #2 . Would prefer to have fixed unless it looks to be overly complicated.

**Comment by ye-luo** [comment]
I have prepared the test case for checking restart. Not integrated into ctest yet.
The electron coordinates and random numbers seem loaded correctly.
1) The extra VMC block (no electron move) and the restart run with the same VMC block agree on KE, ElecElec, LocalECP but NonLocalECP is slightly different. I can't trace all the places where random numbers are used.
2) Run the restart run many times and I got exactly the same results even the initial random seed is not fixed.

So I will set up the checking based on the energy components except NonLocalECP.
